paragon/ui/controllers/map_cell.py

```diff
@@ -152,3 +151,111 @@ def toggle_mode(self):
+            self.reset_animation()
+        if ev.button() == QtCore.Qt.RightButton:
+            self._show_context_menu(ev)
+
+    def paintEvent(self, event):
+        painter = QPainter(self)
+        if self.sprite and self.sprite.frame_height and self.sprite.frame_width and self.sprite.animation_data:
+            if self.sprite.team == "赤" and self.animation_index in [0, 1]:
+                painter.scale(-self.zoom, self.zoom)
+                draw_pos_x = int((-self.width()/self.zoom - self.sprite.frame_width)/2)
+            else:
+                painter.scale(self.zoom, self.zoom)
+                draw_pos_x = int((self.height()/self.zoom - self.sprite.frame_width)/2)
+            draw_pos_y = int((self.width()/self.zoom - self.sprite.frame_height)/2)
+            frame_width = self.sprite.frame_width
+            frame_height = self.sprite.frame_height
+        elif self.sprite and self.sprite.frame_height and self.sprite.frame_width:
+            painter.scale(self.zoom, self.zoom)
+            draw_pos_x = int((self.height()/self.zoom - self.sprite.frame_width)/2),
+            draw_pos_y = int((self.height()/self.zoom - self.sprite.frame_height)/2),
+            frame_width = self.sprite.frame_width
+            frame_height = self.sprite.frame_height
+        else:
+            painter.scale(self.zoom, self.zoom)
+            draw_pos_x = int((self.width()/self.zoom - 32)/2)
+            draw_pos_y = int((self.height()/self.zoom - 32)/2)
+            frame_width = 32
+            frame_height = 32
+
+        painter.drawPixmap(
+            draw_pos_x,
+            draw_pos_y,
+            self.pixmap(), 
+            self.current_frame.x(),
+            self.current_frame.y(), 
+            frame_width, 
+            frame_height
+        )
+        painter.end()
+
+class FE14MapCell(MapCell, FE14UnitSpriteItem):
+    def __init__(self, row, column, sprite_svc):
+        super().__init__(row, column, sprite_svc)
+        self.new_animation.connect(self.draw_new_animation)
+        self.reset_animation_to_idle.connect(self.idle_animation)
+
+    def mousePressEvent(self, ev: QMouseEvent):
+        super().mousePressEvent(ev)
+        if ev.button() == QtCore.Qt.LeftButton:
+            self.reset_animation()
+        if ev.button() == QtCore.Qt.RightButton:
+            self._show_context_menu(ev)
+
+    def paintEvent(self, event):
+        painter = QPainter(self)
+
+        if self.sprite and self.sprite.animation_data:
+            frame_width = self.sprite.animation_data[self.animation_index].frame_data[self.frame_index].body_width
+            frame_height = self.sprite.animation_data[self.animation_index].frame_data[self.frame_index].body_height
+            draw_pos_y = int((self.height()/self.zoom - frame_height)/2) + self.sprite.animation_data[self.animation_index].frame_data[self.frame_index].body_offset_y
+
+            if self.sprite.team  in ["赤", "紫"] and self.animation_index == 0:
```

Comment:
Nitpick: spacing around the and.

Also, since this is FE13 sprite service... maybe make the team check a function in the service so it's easier to understand what this is doing?